File: go-client-sdk/pkg/models/shared/codatpublicapimodelsfilemetadatamodel.go
package shared

import (
	"time"
)

type CodatPublicAPIModelsFileMetadataModel struct {
	DisplayName *string    `json:"displayName,omitempty"`
	FileName    *string    `json:"fileName,omitempty"`
	SourceType  *string    `json:"sourceType,omitempty"`
	Uploaded    *time.Time `json:"uploaded,omitempty"`
}

File: typescript-client-sdk/src/sdk/models/shared/codatpublicapimodelsfilemetadatamodel.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { Expose, Transform } from "class-transformer";


export class CodatPublicApiModelsFileMetadataModel extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "displayName" })
  displayName?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "fileName" })
  fileName?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "sourceType" })
  sourceType?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "uploaded" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  uploaded?: Date;
}